Gems/Atom/Feature/Common/Code/Source/CoreLights/ShadowmapPass.cpp

```diff
@@ -140,10 +157,62 @@ namespace AZ
             m_scissorState = scissor;
         }
 
+        void ShadowmapPass::SetClearShadowDrawPacket(AZ::RHI::ConstPtr<RHI::DrawPacket> clearShadowDrawPacket)
+        {
+            m_clearShadowDrawPacket = clearShadowDrawPacket;
+            m_clearShadowDrawItemProperties = clearShadowDrawPacket->GetDrawItem(0);
+        }
+
         void ShadowmapPass::UpdatePipelineViewTag(const RPI::PipelineViewTag& viewTag)
         {
             SetPipelineViewTag(viewTag);
         }
+        
+        void ShadowmapPass::SetupFrameGraphDependencies(RHI::FrameGraphInterface frameGraph)
+        {
+            Base::SetupFrameGraphDependencies(frameGraph);
+
+            // Override the estimated item count to be what the raster pass would report + 1.
+            frameGraph.SetEstimatedItemCount(static_cast<uint32_t>(m_drawListView.size()) + 1);
+        }
+
+        void ShadowmapPass::SubmitDrawItems(const RHI::FrameGraphExecuteContext& context, uint32_t startIndex, uint32_t endIndex, uint32_t offset) const
+        {
+            if (m_isStatic && !m_forceRenderNextFrame)
+            {
+                const auto& views = m_pipeline->GetViews(GetPipelineViewTag());
+                RHI::DrawListView viewDrawList;
+                if (!views.empty())
+                {
+                    const RPI::ViewPtr& view = views.front();
+                    if (view && (view->GetOrFlags() & m_casterMovedBit.GetIndex()) == 0)
+                    {
+                        // Shadow is static and no casters moved since last frame.
+                        return;
+                    }
+                }
+            }
+            m_forceRenderNextFrame = false;
+
+            if (m_clearShadowDrawPacket)
+            {
+                if (startIndex == 0)
+                {
+                    RHI::CommandList* commandList = context.GetCommandList();
+                    commandList->Submit(*m_clearShadowDrawPacket->GetDrawItem(0).m_item, 0);
+                }
+                else
+                {
+                    // Only decrement startIndex if startIndex is greater than 0. This means that RasterPass will submit one less
+                    // draw item for the first batch since the clear draw was submitted in this batch.
+                    --startIndex; 
+                }
+                // RasterPass's draw item indices need to be adjusted to make sure it stays inside its expected bounds.
+                --endIndex;
+                ++offset;
+            }
+            Base::SubmitDrawItems(context, startIndex, endIndex, offset);
```

Comment:
This is the result of some strict validation that is done on the command list where you need to pass the index of the thing you're submitting purely for validation purposes.

RasterPass handles most of the submission. Previously it was basically impossible to add your own draw items in a sub class, but I made changes to make this possible.  `BuildCommandListInternal()` now calls `SubmitDrawItems()` which can be overridden.

So, the clear item _is_ being submitted here as the first item - `commandList->Submit(*m_clearShadowDrawPacket->GetDrawItem(0).m_item, 0);` However, the indices need to be adjusted for the base class so that it submits the correct items with the correct indices and doesn't go out of bounds (ShadowmapPass sets the estimated item count to +1 when there's a clear draw, which the raster pass isn't aware of).

Longer term I think we should relax the validation to make this less ugly. There used to be no validation at all, and it was easy for people to not realize that `BuildCommandListInternal()` could be called multiple times per pass with different submit ranges. However, I think just making sure that you submit the correct number of items should be enough, making sure to index them all seems overkill.